gopherage/pkg/cov/junit/calculation/coveragelist.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package calculation
+
+import (
+	"fmt"
+	"path"
+	"strings"
+)
+
+// CoverageList is a collection and summary over multiple file Coverage objects
+type CoverageList struct {
+	*coverage
+	Group            []coverage
+	summaryCompleted bool
+}
+
+// CovList constructs new (file) Group Coverage
+func newCoverageList(name string) *CoverageList {
+	return &CoverageList{
+		coverage:         &coverage{Name: name},
+		Group:            []coverage{},
+		summaryCompleted: false,
+	}
+}
+
+// Percentage returns the percentage of statements covered
+func (covList *CoverageList) Percentage() string {
+	return fmt.Sprintf("%.1f", covList.Ratio()*100)
+}
+
+// Ratio summarizes the list of coverages and returns the summarized ratio
+func (covList *CoverageList) Ratio() float32 {
+	covList.Summarize()
+	return covList.coverage.Ratio()
+}
+
+// Summarize summarizes all items in the Group and stores the result
+func (covList *CoverageList) Summarize() {
```

Comment:
Does this need to be exported? If it's just an implementation detail of `Ratio`, it shouldn't be.

I'm a little concerned about the caching approach here. Once `Ratio` is called, its results are frozen forever, even if the coverage list should happen to change. At the very least, this should be clearly documented.

That said, I think it might make more sense to remove the caching, and instead cache `Ratio()` at the caller where necessary. For this to make sense you might have to pull `Percentage` out elsewhere, but I think that's probably also fine anyway: formatting a number has little to do with calculating it, and it should probably just be done in  package `junit`, where it's needed.